From 63b71fc889298df89c5f1baa56accee18024b074 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Tue, 23 Feb 2016 10:32:10 +0800
Subject: [PATCH 4/7] atomisp2: keep enabling XNR for ISP2.7 still pipe

ISP2.7 still pipe requires enable_xnr must be true, otherwise
the pipeline can not be created.

Change-Id: I3500e96e30059de8b1b6ac9d5b643d5a55ec2def
Original-Tracked_On: #H1604086285
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/103252
Build: Dessert M BXT RVP (SC)
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Build: Dessert M CHT FFD (SC)
Reviewed-by: DaleiX Liu <daleix.liu@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-21584
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/475696
---
 .../atomisp2/atomisp_driver/atomisp_compat_css20.c | 22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index db33f502..b3825e5 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -729,6 +729,10 @@ static void __apply_additional_pipe_config(
 			/* ISP2.7 does not support DZ, we should always turn
 			   it off. */
 			stream_env->pipe_configs[pipe_id].enable_dz = false;
+			/* ISP2.7 still pipe requires enable_xnr to be true */
+			stream_env->pipe_configs[pipe_id]
+				.default_capture_config.enable_xnr = true;
+			asd->params.capture_config.enable_xnr = true;
 		} else {
 			stream_env->pipe_configs[pipe_id].isp_pipe_version =
 				SH_CSS_ISP_PIPE_VERSION_2_2;
@@ -3464,10 +3468,20 @@ int atomisp_css_exp_id_unlock(struct atomisp_sub_device *asd, int exp_id)
 int atomisp_css_capture_enable_xnr(struct atomisp_sub_device *asd,
 				   bool enable)
 {
-	asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL]
-		.pipe_configs[IA_CSS_PIPE_ID_CAPTURE]
-		.default_capture_config.enable_xnr = enable;
-	asd->params.capture_config.enable_xnr = enable;
+	if (asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL]
+		.pipe_configs[IA_CSS_PIPE_ID_CAPTURE].isp_pipe_version ==
+		IA_CSS_PIPE_VERSION_2_7) {
+		/* ISP2.7 still pipe requires enable_xnr must be true */
+		asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL]
+			.pipe_configs[IA_CSS_PIPE_ID_CAPTURE]
+			.default_capture_config.enable_xnr = true;
+		asd->params.capture_config.enable_xnr = true;
+	} else {
+		asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL]
+			.pipe_configs[IA_CSS_PIPE_ID_CAPTURE]
+			.default_capture_config.enable_xnr = enable;
+		asd->params.capture_config.enable_xnr = enable;
+	}
 	asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL]
 		.update_pipe[IA_CSS_PIPE_ID_CAPTURE] = true;
 
-- 
1.9.1

